lib/arjdbc/vertica/adapter.rb

```diff
@@ -67,12 +74,34 @@ module Vertica
     }
 
     TIMESTAMP_COLUMNS = [
-      "created_at",
-      "created_on",
-      "updated_at",
-      "updated_on"
+        "created_at",
+        "created_on",
+        "updated_at",
+        "updated_on"
     ]
 
+    def columns(table_name, name = nil)
+      sql = "SELECT * from V_CATALOG.COLUMNS WHERE table_name = '#{table_name}';"
+      raw_columns = execute(sql, name || "SCHEMA")
+
+      # Vertica doesn't use the type name of bigint since all ints are 8-byte, so when the type is int use bigint instead
+
+      columns = raw_columns.map do |raw_column|
+        ::ActiveRecord::ConnectionAdapters::VerticaColumn.new(
+            raw_column['column_name'],
+            raw_column['column_default'],
+            raw_column['data_type_id'],
+            raw_column['table_name'],
+            raw_column['data_type'] == 'int' ? 'bigint' : raw_column['data_type'],
+            fetch_type_metadata(raw_column['data_type'] == 'int' ? 'bigint' : raw_column['data_type']),
+            raw_column['is_nullable'],
+            raw_column['is_identity']
+        )
+      end
+
+      return columns
```

Comment:
You can probably drop this return and the `columns = ` above since all you need is the map.